client/src/components/settings/toggle-radio-setting.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import '../helpers/toggle-button.css';
+import './toggle-setting.css';
+
+type ToggleRadioSettingProps = {
+  action: string;
+  explain?: string;
+  flag: boolean;
+  flagName: string;
+  toggleFlag: () => void;
+  offLabel: string;
+  onLabel: string;
+};
+
+export default function ToggleRadioSetting({
+  action,
+  explain,
+  flag,
+  flagName,
+  toggleFlag,
+  ...restProps
+}: ToggleRadioSettingProps): JSX.Element {
+  const randomId = Math.floor(Math.random() * 1_000_000);
+  const firstRadioId = `radioA${randomId}`;
+  const secondRadioId = `radioB${randomId}`;
+
+  return (
+    <div className='toggle-setting-container'>
+      <fieldset
+        {...(explain && {
+          'aria-labelledby': `legend${randomId} desc${randomId}`
+        })}
+      >
+        <legend
+          className='sr-only'
+          {...(explain && { id: `legend${randomId}` })}
+        >
+          {action}
+        </legend>
+        <div className='toggle-description'>
+          <p aria-hidden={true}>{action}</p>
+          {explain ? <p id={`desc${randomId}`}>{explain}</p> : null}
+        </div>
+        <div className='toggle-radio-group'>
+          <label htmlFor={firstRadioId} data-checked={flag}>
```

Comment:
```suggestion
          <label htmlFor={flagName} data-checked={flag}>
```

Reply:
Leaving this as `firstRadioId` since it is used in more than one place.